File: DSA/Day43_q2.cpp
class Solution {
public:
    int numTimesAllBlue(vector<int>& flips) {
        int n=flips.size();
        int ans=0, mx=INT_MIN;
        for(int i = 0; i < n; i++) {
            int a=flips[i];
            mx=max(mx, a);
            if(mx<=i+1){
                ans++;
            }
        }
        return ans;
    }
};

File: DSA/Shruti-Pandey/Day_43_q2.java
class Solution {
    public int numTimesAllBlue(int[] flips) {
        int moves=0;
        int right = 0; 
        for(int i=0; i<flips.length; i++){
            right = Math.max(right, flips[i]);
            if(right == i+1){
                moves++;
            }
        }
        return moves;
    }
}
